### PR TITLE
Refactor Frontend and some fixes for dashboard pages

### PR DESCRIPTION
**Pull request by ThatGirlJam**
Big structural changes! Components have now been placed in their respective features. The following describes the main refactorings and fixes

Refactoring:

- Each feature has their own components and hooks folders. Types, assets etc. can be added as folders if necessary as well.
- Each feature has a "index.ts" file. This file is used to export all components / hooks / types that are used by either other features or pages. This is to reduce the lengthiness of import code. [E.g. instead of import EditModal from "../../features/questions/components/EditModal" you can now write import {EditModal} from "../../features/questions"]
- ****Remember to update the "index.ts" file to export new stuff and remove old ones!****
- Different Navbar variations have been added under the shared components folder. Auth --> the one for Login/Register Page. Profile -> profile page etc...
- I have factored out some of the reused components and restyled the buttons and some pages to make it look nicer. [**Haven't done the profile picture page yet, but can restyle it later if want/need be** --> also reminder to self to possibly have a standard navbar for others to copy if possible?]
- Navbars and the buttons on navbars have been standardised. [**will standardise smaller buttons and other reused stuff later on if need be**]
- Clicking on the PeerPrep logo will return user to dashboard (if logged in) or landing page (if not logged in)

Fixes:

- Previously there were 2 pages for dashboard and dashboardForUsers. This was removed as it would allow users to access the dashboard page for admins by simply typing "/dashboard" as the url --> Now there is only one dashboard page, that will display different navbars and different dashboards for users and admins. This separates the concerns and allows users and admins to have different dashboards entirely! [Segregation is done on the components level]

### CODE REVIEW

File: frontend/src/components/navbars/AdminNavBar.tsx [hunk 1]
```diff
@@ -19,22 +21,23 @@ const AdminNavBar: React.FC<AdminNavBarProps> = ({ fetchData }) => {
...
           <div className="container text-off-white">
-            <button
+            {/* <button
               onClick={() => openAddModal()}
               className="bg-green rounded-[25px] p-4 text-2xl hover:bg-emerald-700"
             >
               Add question
-            </button>
+            </button> */}
+            <StandardBigButton
```
Comment (seanpzk) [inline frontend/src/components/navbars/AdminNavBar.tsx:36]: Thanks for these small detailed changes!

**Review by seanpzk — APPROVED** [review]
LGTM, thanks sm for refactoring and even tidying up some of the codestyle :)

**Review by HmuuMyatMoe — APPROVED** [review]
LGTM! Thanks for making it so neat!

**Review by DonovanJJ — APPROVED** [review]
Thanks for refactoring the frontend code! Didn't know you could simplify imports like this 👍 👍

**Pull request merged**